### DISCUSSION TITLE
When hitting enter on a line containing nothing but a comment, make the next line have a comment

### DISCUSSION THREAD

**Discussion by jonathansekela** [godotengine/godot-proposals#11129] [Editor]
Currently, when creating comments in scripts, to create a block of single-line comments, the user needs to manually hit the '#' key every time they go to a new line. The following is a visual example of my proposed enhancement.

We start with a standard "comments before the function" scenario.
![initialcomment](https://github.com/user-attachments/assets/01602f5a-14c9-4ecd-8934-0ee961e28fb8)

Hitting enter on line 9 causes this to happen:
![currentfunctionality](https://github.com/user-attachments/assets/41b7c94c-8f75-439e-a973-5dcda1e875da)

I'd like this to happen instead:
![desiredfunctionality](https://github.com/user-attachments/assets/81baf292-7aef-473f-855d-c5edec6972f1)

Specifically, when hitting enter in the script editor when the cursor is on a line containing nothing but a comment, I'd like the new line to start out being a comment automatically.

**Comment by sortizu** [comment]
I think we should expect the same behavior that is common in other programs like VS Code or Android Studio. From my experience, I haven't use any IDE or code editor with a feature like this, but I can think in some situations where this could be problematic. For example, if someone wanted to add new lines between functions and comments, trying to add more space to improve redability or organization (which I think is a common task) or any other reason, this feature would inadvertently create extra comment lines. This forces the user to move the caret to the start of the next line manually to insert a new line as intended, which can be quite annoying.
However, I like the idea of automatically creating a new comment line when hitting enter **inside** an existing comment, since the most common reason to do this is when your are formatting comments across multiple lines. This was proposed in #4955, sadly it didn't progress any further.

**Reply by jonathansekela** [reply]
I'm coming from the perspective of applications like Visual Studio and the Jetbrains IDEs. In those, this functionality happens automatically. Not text editors like VS Code, but full IDEs. I guess it's a question of whether Godot's script editor should behave like a text editor or like an IDE. I would argue that despite its GUI and node-based development process, Godot is more of a full development suite, and the editor itself should reflect that.

**Reply by sortizu** [reply]
Yes, I totally agree that Godot should be treated as an IDE, and it should offer integrated features to be as useful as other IDEs. I think this feature is more related to usability. Since I haven't used Visual Studio for a while, or any other program that implements this by default, I can't really see how helpful it can be. Now, speaking entirely from my perspective, I find this behavior very annoying in some usual situations. Perhaps it’s a subjective preference, but as I see it right now it makes you consider that there is more than one behavior to expect when you try to add new lines, no matter in which section or line of the code you are, and even when you dont want that to happen. So I think this should be enabled only when you indicate that you want that. Therefore, I'm more interested in proposals like #8244 because they can be considered like an special type of comments. Another option could be to use accesible shortcuts and easy to remember, something like shift + Enter.

**Comment by sockeye-d** [comment]
I don't agree with this for single line comments, but it's pretty normal ide behavior for documentation comments (those that start with `##`). It would also be nice if it retained the indentation

**Comment by dalexeev** [comment]
See also:

* #4955
* #8244
* https://github.com/godotengine/godot/pull/95955

**Reply by jonathansekela** [reply]
whoops, I thought I'd looked for this one before posting it but turns out I didn't look very well. My bad - should I delete this discussion and just comment on these ones then?

**Reply by tetrapod00** [reply]
For future reference, there is no need to delete this. *Deleting* things is only done for actual spam and the like. Duplicate issues are marked "archived" and linked to the original, not deleted. 

Also, this is a discussion not an issue, so it's considered in a separate set of things. It will just naturally be scrolled away by new discussions.

You should comment on the existing issues if have something to add to them.
